server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -1348,8 +1372,10 @@ public void onSuccess(String source) {
                                         }
                                     }
                                     lagDetector.startLagDetector(publishRequest.getAcceptedState().version());
+                                    logIncompleteNodes();
```

Comment:
Because `10s` divides `30s` exactly, there is a chance that by default sometimes we'll log the third periodic warning and then the end-of-publication warning in quick succession. We could just not log the end-of-publication warning here.

Reply:
@DaveCTurner that's not so bad imo, we'll see the thread this was logged from in the logs as well so it's not that weird? (put differently, it doesn't bother me :P and maybe by some random chance this will help debug some locking issue with `mutex` who knows :D)

Reply:
Thinking about this made me slightly doubt whether we should do a periodic logging here based on an interval.  I wonder if we should instead just allow to define two timeouts, one for debug (5s) and one for info logging (10s), and then do warn logging upon publication timeout (i.e. what we have here).

Reply:
I think an earlier debug log is of limited value, because people don't often run with debug logging on the offchance that it catches something interesting like this. I do agree that it seems sufficient to emit a single info message after 10 seconds, and then a warning after 30 seconds, so I've implemented that in 16fae080007.